src/tera.rs

```diff
@@ -306,11 +306,39 @@ impl Tera {
         renderer.render()
     }
 
+    /// Renders a one off template (for example a template coming from a user
+    /// input) given a `Context` and an instance of Tera. This allows you to
+    /// render templates using custom filters or functions. **Warning** This
+    /// will add a template called `__one_off` to your instance. This method is
+    /// mainly useful for testing.
+    ///
+    /// Any errors will mention the `__one_off` template: this is the name given to the template by
+    /// Tera
+    ///
+    /// ```rust,ignore
+    /// let mut context = Context::new();
+    /// context.insert("greeting", &"hello");
+    /// Tera::one_off("{{ greeting }} world", &context, true);
+    /// ```
+    pub fn render_str(
+        &mut self,
+        input: &str,
+        context: &Context,
+        autoescape: bool,
+    ) -> Result<String> {
+        self.add_raw_template("__one_off", input)?;
+        if autoescape {
+            self.autoescape_on(vec!["__one_off"]);
+        }
+
+        self.render("__one_off", &context)
```

Comment:
That function has the side effect of changing the auto-escape of the whole instance.
If that function is only really used for testing in practice, maybe we can just clone the Tera instance and change auto-espacing/adding the template on that new instance that gets dropped at the end of the function.